integration-cli/docker_cli_swarm_test.go

```diff
@@ -95,6 +95,13 @@ func (s *DockerSwarmSuite) TestSwarmInitUnspecifiedAdvertiseAddr(c *check.C) {
 	c.Assert(out, checker.Contains, "advertise address must be a non-zero IP address")
 }
 
+func (s *DockerSwarmSuite) TestSwarmInitUnspecifiedDataPathAddr(c *check.C) {
+	d := s.AddDaemon(c, false, false)
+	out, err := d.Cmd("swarm", "init", "--datapath-addr", "0.0.0.0")
+	c.Assert(err, checker.NotNil)
+	c.Assert(out, checker.Contains, "data path address must be a non-zero IP")
```

Comment:
Perhaps we could also check that a `--datapath-addr` argument with a port number is rejected.